Bindings commercetools

harrykimpel opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding documentation for the commercetools output binding.

## Issue reference

#2351

harrykimpel committed:
Adding commercetools binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel committed:
Adding commercetools binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel committed:
Fixing GitHub Codespaces link validation issue

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel commented:
the validate-links checker fails for the GitHub Codespaces overview page with a 403

berndverst commented on daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md:
```diff
@@ -0,0 +1,71 @@
+  type: bindings.commercetools
+  version: v1
+  metadata:
+  - name: Region # required.
```
Comparing to your code PR all these meatadata fields must start with a lowercase character

yaron2 replied:
ping @harrykimpel

harrykimpel replied:
thanks @yaron2, will update

harrykimpel replied:
done

yaron2 committed:
Merge branch 'v1.8' into bindings-commercetools

harrykimpel committed:
Updated docs for commercetools binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel committed:
Merge branch 'bindings-commercetools' of https://github.com/harrykimpel/docs into bindings-commercetools

harrykimpel committed:
Updated docs for commercetools binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

greenie-msft commented:
Hi @harrykimpel,

It looks like some additional files were checked-in unintentionally. Can you please revert those files (dccache ?)

Also, I noticed you added a link to a sample in your GitHub account. Are you open to submitting that to the official Samples repo in the Dapr org?

harrykimpel committed:
Updated docs for commercetools binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel commented:
> Hi @harrykimpel,
> 
> It looks like some additional files were checked-in unintentionally. Can you please revert those files (dccache ?)
> 
> Also, I noticed you added a link to a sample in your GitHub account. Are you open to submitting that to the official Samples repo in the Dapr org?

@greenie-msft I removed the dccache file. I am also happy to submit the sample to the official sample repo. I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?

msfussell commented:
>> I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?

Yes, submit to this Samples repo. The expectation on this repo is that the submitted chooses to keep the sample current and maintained. If these go very stale over time (i.e. Feature removed or does not compile) they get removed. 

Please update this link in the doc- [Sample app](https://github.com/harrykimpel/dapr-commercetools-sample) that leverages the commercetools binding with sample GraphQL query and then resubmit.

msfussell committed:
Merge branch 'v1.8' into bindings-commercetools

harrykimpel committed:
Adding sample for commercetools GraphQL binding

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel committed:
Merge branch 'bindings-commercetools' of https://github.com/harrykimpel/docs into bindings-commercetools

harrykimpel commented:
> > > I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?
> 
> Yes, submit to this Samples repo. The expectation on this repo is that the submitted chooses to keep the sample current and maintained. If these go very stale over time (i.e. Feature removed or does not compile) they get removed.
> 
> Please update this link in the doc- [Sample app](https://github.com/harrykimpel/dapr-commercetools-sample) that leverages the commercetools binding with sample GraphQL query and then resubmit.

@msfussell I added the samples app to the samples repo and updated the link in the docs

harrykimpel committed:
Updated docs for Azure SQL database

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel commented:
Tried to update the link to the Azure SQL DB docs, but now I am getting a fail for a localized version

msfussell commented:
@harrykimpel - See comment on the Sample repo PR here https://github.com/dapr/samples/pull/106

msfussell committed:
Merge branch 'v1.8' into bindings-commercetools

berndverst commented on daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md:
```diff
@@ -0,0 +1,71 @@
+
+| Field              | Required | Binding support |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| Region | Y | Output | The region of the commercetools project | `"europe-west1"` |
```
These fields should all start with a lowercase character to match the code PR:

https://github.com/dapr/components-contrib/pull/1652/files

harrykimpel replied:
@berndverst done

harrykimpel committed:
Updating commercetools docs

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

harrykimpel committed:
Merge branch 'bindings-commercetools' of https://github.com/harrykimpel/docs into bindings-commercetools

harrykimpel commented:
> @harrykimpel - See comment on the Sample repo PR here [dapr/samples#106](https://github.com/dapr/samples/pull/106)

@msfussell done

harrykimpel committed:
Updating commercetools docs

Signed-off-by: Harry Kimpel <harrykimpel@hotmail.com>

msfussell committed:
Merge branch 'v1.8' into bindings-commercetools

msfussell reviewed (approved):
LGTM. Thanks for the contribution!

msfussell commented on daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md:
```diff
@@ -0,0 +1,71 @@
+- [How-To: Trigger application with input binding]({{< ref howto-triggers.md >}})
+- [How-To: Use bindings to interface with external resources]({{< ref howto-bindings.md >}})
+- [Bindings API reference]({{< ref bindings_api.md >}})
+- [Sample app](https://github.com/dapr/samples/commercetools-graphql-sample) that leverages the commercetools binding with sample GraphQL query
```
Given I have merged the Sample into the samples repo, the URL for this is https://github.com/dapr/samples/tree/master/commercetools-graphql-sample

Can you confirm this is your expectation, and then will have to change this

```suggestion
- [Sample app](https://github.com/dapr/samples/tree/master/commercetools-graphql-sample) that leverages the commercetools binding with sample GraphQL query
```

msfussell commented:
@harrykimpel - Just found the link issue above. PTAL?

harrykimpel committed:
Update daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md

yes, fine with me

Co-authored-by: Mark Fussell <markfussell@gmail.com>

msfussell committed:
Merge branch 'v1.8' into bindings-commercetools

msfussell commented:
Thanks for the amazing contribution!!
